cachi2/core/utils.py

```diff
@@ -197,6 +198,12 @@ def get_cache_dir() -> Path:
     return cache_dir.joinpath("cachi2")
 
 
-def first(predicate: Callable, iterable: Iterable, fallback: Any) -> Any:
+def first_for(predicate: Callable, iterable: Iterable, fallback: Any) -> Any:
```

Comment:
Sounds like this change should happen in the commit which is introducing `first` as utility function rather than renaming it later in the series.